vue/sbc-common-components/src/services/keycloak.services.ts

```diff
@@ -0,0 +1,160 @@
+import Keycloak, { KeycloakInitOptions, KeycloakInstance, KeycloakLoginOptions, KeycloakTokenParsed } from 'keycloak-js'
+import { SessionStorageKeys } from '../util/constants'
+import { UserInfo } from '../models/userInfo'
+import ConfigHelper from '../util/config-helper'
+
+interface UserToken extends KeycloakTokenParsed {
+  lastname: string;
+  firstname: string;
+  email: string;
+  // eslint-disable-next-line camelcase
+  realm_access_roles: string[];
+  jti: string;
+  username: string;
+}
+
+class KeyCloakService {
+  private kc: KeycloakInstance
+  private parsedToken: any
+  private static instance: KeyCloakService
+
+  public static getInstance (): KeyCloakService {
+    return (this.instance) ? this.instance : new KeyCloakService()
+  }
+
+  init (idpHint: string) {
+    this.cleanupSession()
+    const token = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakToken)
+    const keycloakConfig = ConfigHelper.getKeycloakConfigUrl()
+    this.kc = Keycloak(keycloakConfig)
+    const kcLogin = this.kc.login
+    this.kc.login = (options?: KeycloakLoginOptions) => {
+      options.idpHint = idpHint
+      return kcLogin(options)
+    }
+    return this.kc.init({ token: token, onLoad: 'login-required' })
+  }
+
+  initSession () {
+    ConfigHelper.addToSession(SessionStorageKeys.KeyCloakToken, this.kc.token)
+    ConfigHelper.addToSession(SessionStorageKeys.KeyCloakIdToken, this.kc.idToken)
+    ConfigHelper.addToSession(SessionStorageKeys.KeyCloakRefreshToken, this.kc.refreshToken)
+    ConfigHelper.addToSession(SessionStorageKeys.UserFullName, this.getUserInfo().fullName)
+    ConfigHelper.addToSession(SessionStorageKeys.UserFullName, this.getUserInfo().fullName)
+    ConfigHelper.addToSession(SessionStorageKeys.UserKcId, this.getUserInfo().keycloakGuid)
+
+    this.parsedToken = this.kc.tokenParsed as UserToken
+
+    // Set flag in session storage so that common components will know account type
+    ConfigHelper.addToSession(SessionStorageKeys.UserAccountType, this.getUserInfo().loginSource)
+  }
+
+  getUserInfo () : UserInfo {
+    if (!this.parsedToken) {
+      this.parsedToken = this.decodeToken()
+    }
+    return {
+      lastName: this.parsedToken.lastname,
+      firstName: this.parsedToken.firstname,
+      email: this.parsedToken.email,
+      roles: this.parsedToken.realm_access.roles,
+      keycloakGuid: this.parsedToken.sub,
+      userName: this.parsedToken.username,
+      fullName: `${this.parsedToken.firstname} ${this.parsedToken.lastname}`,
+      loginSource: this.parsedToken.loginSource
+    }
+  }
+
+  async logout (redirectUrl: string) {
+    let token = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakToken)
+    if (token) {
+      this.kc = Keycloak(ConfigHelper.getKeycloakConfigUrl())
+      let kcOptions :KeycloakInitOptions = {
+        onLoad: 'login-required',
+        checkLoginIframe: false,
+        timeSkew: 0,
+        token: sessionStorage.getItem('KEYCLOAK_TOKEN'),
```

Comment:
Can we use the config helper to retrieve these, just so it's consistent with the rest of the library?  Should be `ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakToken)`

Reply:
sure, will do
Thanks @sutherlanda

Reply:
done